tests/switcheroo/ssh/unit/retriever/test_cli.py

```diff
@@ -21,21 +17,30 @@ def test_local_ds_with_bucket_flag_exception(self, some_name: str):
         ]
         assert_that(
             calling(main).with_args(user_input),  # type: ignore
-            raises(InvalidArgumentError),
+            raises(SystemExit),
         )
+        captured = capsys.readouterr()
+        expected = 'Invalid argument "--bucket" when retrieving the keys locally'
+        assert_that(captured.err, contains_string(expected))
 
-    def test_s3_ds_with_no_bucket_exception(self, some_name: str):
-        "Test that exception is thrown when the user doesn't input --bucket with s3 ds"
+    def test_s3_ds_with_no_bucket(self, capsys, some_name: str):
+        "Test that erorr is raised when the user doesn't input --bucket with s3 ds"
         user_input = ["-ds", "s3", f"{some_name}"]
         assert_that(
             calling(main).with_args(user_input),  # type: ignore
-            raises(MissingArgumentError),
+            raises(SystemExit),
         )
+        captured = capsys.readouterr()
+        expected = "The s3 option requires a specified bucket name!"
+        assert_that(captured.err, contains_string(expected))
 
-    # def test_no_user_exception(self):
-    #     "Test that exception is thrown when no user is input"
-    #     user_input = [
-    #         "-ds",
-    #         "local",
-    #     ]
-    #     assert_that(calling(main).with_args(user_input), raises(SystemExit))
+    def test_no_user_exception(self, capsys):
+        "Test that error is raised when no user is input"
+        user_input = [
+            "-ds",
+            "local",
+        ]
+        assert_that(calling(main).with_args(user_input), raises(SystemExit))
+        captured = capsys.readouterr()
+        expected = "the following arguments are required: user"
+        assert_that(captured.err, contains_string(expected))
```

Comment:
Honestly I don't think it's necessary to test that argparse works as expected, personally. Then there's no end to the tests we could write